### PR TITLE
feat: add newsletter subscription button to footer

### PR DESCRIPTION
**Pull request by burden**
- Adds a new `showNewsletterButton` prop to `SysFooterbar` 
- When `showNewsletterButton` is `true` a section with a "Subscribe" button is shown.
- When the Subscribe button is pressed, an `action` event will be emitted.

### CODE REVIEW

File: src/components/sys-footerbar.vue [hunk 1]
```diff
@@ -104,62 +104,79 @@
...
+          <div>
+            <div
+              v-if="showNewsletterButton"
+              :class="$style.subscribe"
+            >
+              <p>Subscribe for updates on products, OS features, and promotions.</p>
+              <p>
+                <button
+                  class="button"
+                  @click.prevent="$emit('action')"
```
Comment (btkostner) [inline src/components/sys-footerbar.vue:117]: I'd probably change the event name to something like `subscribe`. `action` is pretty generic and doesn't say what kind of thing actually happened

File: package.json [hunk 2]
```diff
@@ -50,64 +50,59 @@
   "publishConfig": {
     "access": "public"
   },
-  "dependencies": {
-    "@fortawesome/free-brands-svg-icons": "^5.13.0",
-    "@fortawesome/free-solid-svg-icons": "^5.12.1",
-    "@fortawesome/vue-fontawesome": "^0.1.9",
-    "@system76/design": "^5.2.1"
-  },
```
Comment (btkostner) [inline package.json:58]: These should still be dependencies. Not sure how that got switched.

File: test/browser/sys-footerbar/with-button.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { Selector } from 'testcafe'
+
+import { assertScreenshot, WIDTHS } from '../../utility/browser'
+
+fixture`SysFooterbar with button`
+  .page`http://localhost:9476/iframe.html?id=components-sysfooterbar--with-button&viewMode=story`
```
Comment (btkostner) [inline test/browser/sys-footerbar/with-button.js:6]: Sweet!

File: src/components/sys-footerbar.vue [hunk 4]
```diff
@@ -269,14 +303,54 @@
...
+    transition-duration: 100ms;
+    transition-property: background-color, border-color, box-shadow;
+    transition-timing-function: ease;
+    user-select: none;
+    letter-spacing: normal;
+    font-family: var(--font-family-sans);
+    text-transform: uppercase;
+    font-size: 1rem;
+    padding: 0.4em 1.6em;
+  }
```
Comment (btkostner) [inline src/components/sys-footerbar.vue:333]: Redefining styles kinda sucks, but we have to do that here. Hopefully down the line we can change the button to blend in a little bit better with the footer.

**Review by btkostner — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (burden) [reply package.json:58]: I had to run an `npm update` which is likely what happened here. Can I add these back manually to fix it?

**Comment by burden** [comment]
[Newsletter improvements](https://trello.com/c/ZCgVctPv/641-newsletter-improvements)

File: package.json (hunk 2 above)
Comment (burden) [reply package.json:58]: @btkostner does this need to be fixed?

Comment (btkostner) [reply package.json:58]: You should be able to just move these back to `dependencies` and run `npm i`

**Review by btkostner — APPROVED** [review]

**Pull request merged**